src/test/java/com/baidu/hugegraph/functional/HugeClientHttpsTest.java

```diff
@@ -0,0 +1,186 @@
+                           .build();
+        Assert.assertTrue(client.graphs().listGraph().contains("hugegraph"));
+        testHttpsAddVertexPropertyValue();
+    }
+
+    @Test
+    public void testHttpsClientNewBuilderWithConnectionPool() {
+        client = HugeClient.builder(BASE_URL, GRAPH)
+                           .configUser(USERNAME, PASSWORD)
+                           .configTimeout(TIMEOUT)
+                           .configPool(MAX_CONNS, MAX_CONNS_PER_ROUTE)
+                           .configSSL(PROTOCOL, TRUST_STORE_FILE,
+                                      TRUST_STORE_PASSWORD)
+                           .configIdleTime(IDLE_TIME)
+                           .build();
+        Assert.assertTrue(client.graphs().listGraph().contains("hugegraph"));
+        testHttpsAddVertexPropertyValue();
+    }
+
+    @Test
+    public void testHttpsClientNewBuilderZeroPoolParam() {
+        client = HugeClient.builder(BASE_URL, GRAPH)
+                           .configUser(USERNAME, PASSWORD)
+                           .configTimeout(TIMEOUT)
+                           .configPool(0, 0)
+                           .configSSL(PROTOCOL, TRUST_STORE_FILE,
+                                      TRUST_STORE_PASSWORD)
+                           .build();
+        Assert.assertTrue(client.graphs().listGraph().contains("hugegraph"));
+        testHttpsAddVertexPropertyValue();
+    }
+
+    @Test
+    public void testHttpsClientBuilderWithConnectionPoolNoParam() {
+        Assert.assertThrows(IllegalArgumentException.class, () -> {
+            HugeClient.builder(BASE_URL, GRAPH)
+                      .configUrl(null)
+                      .configGraph(null)
+                      .configSSL(null,"","")
+                      .build();
+        }, e -> {
+            Assert.assertContains("The url parameter can't be null",
+                                  e.getMessage());
+        });
+    }
+
+    @Test
+    public void testHttpsClientBuilderWithConnectionPoolNoGraphParam() {
+        Assert.assertThrows(IllegalArgumentException.class, () -> {
+            HugeClient.builder(BASE_URL, GRAPH)
+                      .configGraph(null)
+                      .configSSL(null, "", "")
+                      .build();
+        }, e -> {
+            Assert.assertContains("The graph parameter can't be null",
+                                  e.getMessage());
+        });
+    }
+
+    @Test
+    public void testHttpsClientBuilderWithConnectionPoolZeroIdleTimeParam() {
+        Assert.assertThrows(IllegalArgumentException.class, () -> {
+            HugeClient.builder(BASE_URL, GRAPH)
+                      .configIdleTime(0)
+                      .build();
+        }, e -> {
+            Assert.assertContains("The idleTime parameter must be > 0,but got",
+                                  e.getMessage());
+        });
+    }
+
+    private void testHttpsAddVertexPropertyValue() {
```

Comment:
prefer to rename to `addVertexAndCheckPropertyValue()`

method name prefix with 'test' seems like one test case and the method content is not related with https at all